hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -2480,26 +2488,45 @@ public void setTimes(OzoneObj obj, String keyName, long mtime, long atime)
     ozoneManagerClient.setTimes(builder.build(), mtime, atime);
   }
 
+  private ExecutorService createThreadPoolExecutor(
+      int corePoolSize, int maximumPoolSize, String threadNameFormat) {
+    return new ThreadPoolExecutor(corePoolSize, maximumPoolSize,
+        60, TimeUnit.SECONDS, new SynchronousQueue<>(),
+        new ThreadFactoryBuilder()
+            .setNameFormat(threadNameFormat)
+            .build(),
+        new ThreadPoolExecutor.CallerRunsPolicy());
+  }
+
   public ExecutorService getECReconstructExecutor() {
-    // local ref to a volatile to ensure access
-    // to a completed initialized object
-    ExecutorService executor = ecReconstructExecutor;
-    if (executor == null) {
+    ExecutorService localRef = ecReconstructExecutor;
+    if (localRef == null) {
       synchronized (this) {
-        executor = ecReconstructExecutor;
-        if (executor == null) {
-          ecReconstructExecutor = new ThreadPoolExecutor(
-              EC_RECONSTRUCT_STRIPE_READ_POOL_MIN_SIZE,
+        localRef = ecReconstructExecutor;
+        if (localRef == null) {
+          localRef = createThreadPoolExecutor(EC_RECONSTRUCT_STRIPE_READ_POOL_MIN_SIZE,
               clientConfig.getEcReconstructStripeReadPoolLimit(),
-              60, TimeUnit.SECONDS, new SynchronousQueue<>(),
-              new ThreadFactoryBuilder()
-                  .setNameFormat("ec-reconstruct-reader-TID-%d")
-                  .build(),
-              new ThreadPoolExecutor.CallerRunsPolicy());
-          executor = ecReconstructExecutor;
+              "ec-reconstruct-reader-TID-%d");
+          ecReconstructExecutor = localRef;
+        }
+      }
+    }
+    return localRef;
+  }
+
+  public ExecutorService getECWriteThreadPool() {
+    ExecutorService localRef = ecWriteExecutor;
+    if (localRef == null) {
+      synchronized (this) {
+        localRef = ecWriteExecutor;
+        if (localRef == null) {
+          localRef = createThreadPoolExecutor(EC_WRITE_POOL_MIN_SIZE,
+              clientConfig.getEcClientWritePoolLimit(),
```

Comment:
One question: should we make this an unbounded cached thread pool?  This would help resource re-use (as opposed to current single thread per output stream), but also allow scaling further without config change.

Reply:
Do you means `Executors.newCachedThreadPool()`? But an unlimited threads likely to consume too much client-side performance; after all, it is possible that a user's client is not necessarily configured to be very strong

Reply:
I was thinking it's the client's decision to start few or many parallel writes.  With the new config, the responsibility is still at the client-side, but now it needs to be configured explicitly.

I don't know if the config helps in practice or not.

Reply:
Understand, in fact, the original `BlockOutputStream` is equivalent to an unbounded thread because a new thread is created for each `BlockOutputStream` instance.